client/v3/watch.go

```diff
@@ -676,6 +677,17 @@ func (w *watchGrpcStream) run() {
 				closeErr = err
 				return
 			}
+			if isUnavailableErr(w.ctx, err) {
+				// retry, but backoff
+				if backoff < maxBackoff {
+					// 25% backoff factor
+					backoff = backoff + backoff/4
+					if backoff > maxBackoff {
+						backoff = maxBackoff
+					}
+				}
+				time.Sleep(backoff)
+			}
```

Comment:
The code is duplicated to https://github.com/etcd-io/etcd/blob/6a0bbf346256960cbbe0218d6ab13443ee93e8e3/client/v3/watch.go#L1018-L1028 

Please get it included in a function so that it can be reused by both places.

Reply:
Thanks for the review. The backoff function has been added.